sjb/hack/determine_install_upgrade_version.py

```diff
@@ -84,9 +98,21 @@ def determine_search_version(pkg_name, pkg_version):
 	yb.conf.disable_excludes = ["all"]
 	sys.stdout = old_stdout
 	available_pkgs = rpmutils.unique(yb.doPackageLists('available', patterns=[pkg_name], showdups=True).available)
-	search_version = determine_search_version(pkg_name, pkg_version)
 	available_pkgs = remove_duplicate_pkgs(available_pkgs)
 	available_pkgs.sort(lambda x, y: rpmutils.compareEVR((x.epoch, x.version, x.release), (y.epoch, y.version, y.release)))
-	matching_pkgs = get_matching_versions(pkg_name, available_pkgs, search_version)
-	install_version = get_install_version(matching_pkgs)
-	print_version_vars(install_version)
\ No newline at end of file
+
+	search_install_version = determine_install_version(pkg_name, pkg_version)
+
+	matching_install_pkgs = get_matching_versions(pkg_name, available_pkgs, search_install_version)
+	install_version = get_last_version(matching_install_pkgs)
+
+	# If dependency target branch is specified for a release or an enterprise release, determine the upgrade version
+	# by listing all the versions which major and minor version match with the release version and from the list pick
+	# the last one. Otherwise upgrade to the version of the input_pkg
```

Comment:
Even if:
OPENSHIFT_ANSIBLE_TARGET_BRANCH = release-1.5
ORIGIN_TARGET_BRANCH = master
in case of
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_openshift_ansible_extended_conformance_install_update/280/parameters/

in this case we can could even skip the building of origin.. right ?